AK/Platform.h

```diff
@@ -34,46 +34,46 @@
 #endif
 
 #if defined(__clang__)
-#    define AK_COMPILER_CLANG
+#    define AK_COMPILER_CLANG 1
```

Comment:
why though?
the existing mechanism worked fine, `1` is meaningless for these

Reply:
The idea behind it is that `#if SOME_VARIABLE` wouldn't work as "intended" if SOME_VARIABLE is defined but empty (whereas `1` would evaluate to be true). However, since we are actually doing compatibility things here and not configuring in some optional features, a missing `def` will be caught relatively quick. The existing mechanism really is more than enough.

Reply:
yeah, I understand that, but I don't understand what's wrong with just `#if defined()`.